testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -1472,4 +1472,30 @@ public boolean isSupportedInTrino()
         CREATE_TABLE_AS_SELECT,
         CREATE_TABLE_WITH_NO_DATA_AND_INSERT,
     }
+
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS}, dataProvider = "storageFormats")
+    public void testSparkReadsTrinoTableAfterVacuum(StorageFormat storageFormat)
+    {
+        String baseTableName = "test_spark_reads_trino_partitioned_table_after_vacuum2" + storageFormat;
+        String trinoTableName = trinoTableName(baseTableName);
+        String sparkTableName = sparkTableName(baseTableName);
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + trinoTableName);
+
+        onTrino().executeQuery(format("CREATE TABLE %s (_string VARCHAR, _bigint BIGINT) WITH (partitioning = ARRAY['_string'], format = '%s')", trinoTableName, storageFormat));
+        onTrino().executeQuery(format("INSERT INTO %s VALUES ('a', 1001), ('b', 1002), ('c', 1003)", trinoTableName));
+        onTrino().executeQuery(format("INSERT INTO %s VALUES ('d', 1004), ('e', 1005), ('f', 1006)", trinoTableName));
+        onTrino().executeQuery(format("INSERT INTO %s VALUES ('g', 1007), ('h', 1008), ('i', 1009)", trinoTableName));
+
+        onTrino().executeQuery("SET SESSION iceberg.vacuum_min_retention = '0s'");
+        onTrino().executeQuery(format("ALTER TABLE %s EXECUTE VACUUM (retention_threshold => '0s')", trinoTableName));
```

Comment:
Maybe one where the vacuum actually has files to remove? Do an Optimize + Vac

Reply:
In this case it removes old metadata files. But sure I can add test with optimize too